runtime/serde/serde-json/common/test/aws/smithy/kotlin/runtime/serde/json/JsonDeserializerIgnoresKeysTest.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.serde.json
+
+import aws.smithy.kotlin.runtime.serde.SdkFieldDescriptor
+import aws.smithy.kotlin.runtime.serde.SdkObjectDescriptor
+import aws.smithy.kotlin.runtime.serde.SerialKind
+import aws.smithy.kotlin.runtime.serde.deserializeStruct
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class JsonDeserializerIgnoresKeysTest {
+    class IgnoresKeysTest {
```

Comment:
this double-nested class seems unnecessary, I think you can just declare these at the top-level without using a `companion object` for simplicity